.eslintrc.js

```diff
@@ -283,8 +283,9 @@ module.exports = {
           // The meaning of rev vs ref is a common source of confusion.
           // Spelling it out makes it clear.
           rev: { revision: true },
-          // Allow since it's a React term
+          // Allow since they are React terms
           props: false,
+          ref: false,
```

Comment:
This one I'm conflicted with (as opposed to `props`) because it would also allow "git refs", which has been a source of confusion and bugs (see above, "git rev" vs "git ref", "revision" vs "reference" is clearer). I left `ref` out of the React exceptions because it gets autofixed to `reference` (e.g. `const buttonReference = useRef()`), which I think is still descriptive (props->properties is highly ambiguous on the other hand, so the exception makes more sense there). Do you disagree?

Reply:
Yeah, I disagree. `ref` is how React refers to these things (https://reactjs.org/docs/refs-and-the-dom.html). The name obviously came from the word `reference`, but `ref` means something slightly different and more specific, just like `props`.

Reply:
I understand that it's a specific React term too, but how do you weigh it against the ambiguity of git rev vs git ref (which has caused real problems)?

Reply:
My thought is that someone who's confused by rev vs. ref wouldn't be helped by auto-renaming to revision and reference. It may be effective for you to rename props in the web app to make them more clearly named, and to monitor this in code review, but the auto-rename rule just doesn't seem like it will clear up confusion.